File: core/java/src/main/java/hara/lang/protocol/IRealize.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.irealize", name = "IRealize")
public interface IRealize<V> {
  @HaraMethod(value = "realized?", arity = 1)
  boolean isRealized();

  @HaraMethod(value = "realize", arity = 1)
  V realize();
}

File: core/rust/src/lang/protocol/irealize.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.irealize", name = "IRealize")]
pub trait IRealize<V> {
    #[hara_method(value = "realized?", arity = 1)]
    fn is_realized(&self) -> bool;
    #[hara_method(value = "realize", arity = 1)]
    fn realize(&self) -> V;
}
